app/models/api_request.rb

```diff
@@ -0,0 +1,47 @@
+class ApiRequest
+  def query(query)
+    uri = request_uri(query)
+    res = send_api_request(uri)
+    format_response(res)
+  end
+
+  def request_uri(query = nil)
+    URI::HTTPS.build(
+      host: host,
+      path: path,
+      query: query
+    )
+  end
+
+  private
+
+  def format_response(res)
+    {
+      code: res.code,
+      body: JSON.parse(res.body)
+    }.with_indifferent_access
+  end
+
+  # Inheriting class defines
+  def host
+    ""
+  end
+
+  # Inheriting class defines
+  def path
+    ""
+  end
+
+  # NOTE: this will need refactoring if we connect to multiple APIs
+  def send_api_request(uri)
+    req = Net::HTTP::Get.new(uri)
+    req.basic_auth(
+      Rails.configuration.x.hses.api_username,
```

Comment:
[suggestion]if you make these methods then it's future proof, but also fine if that feels like an overoptimization now!